### PR TITLE
BUG: pass 2D ndarray and EA-dtype to DataFrame, closes #12513

### LINKED ISSUE #12513: BUG: DataFrame constructor raises error if specify tz dtype `dtype='datetime64[ns, UTC]'`
#### Code Sample, a copy-pastable example if possible

``` Python
import pandas as pd
import numpy as np
array_dim2 = np.arange(10).reshape((5, 2))
df = pd.DataFrame(array_dim2 , dtype='datetime64[ns, UTC]') # doesn't work
```

The error:

``` Python
TypeError                                 Traceback (most recent call last)
<ipython-input-4-7101cf798aa3> in <module>()
----> 1 df = pd.DataFrame(array_dim2 , dtype='datetime64[ns, UTC]')

C:\D\Projects\Github\pandas\pandas\core\frame.py in __init__(self, data, index,
columns, dtype, copy)
    252             else:
    253                 mgr = self._init_ndarray(data, index, columns, dtype=dty
pe,
--> 254                                          copy=copy)
    255         elif isinstance(data, (list, types.GeneratorType)):
    256             if isinstance(data, types.GeneratorType):

C:\D\Projects\Github\pandas\pandas\core\frame.py in _init_ndarray(self, values,
index, columns, dtype, copy)
    412
    413         if dtype is not None:
--> 414             if values.dtype != dtype:
    415                 try:
    416                     values = values.astype(dtype)

TypeError: data type not understood
```
#### Expected Output

``` Python
In [5]: df = pd.DataFrame(array_dim2 , dtype='datetime64[ns, UTC]')

In [6]: df
Out[6]:
                              0                                           1
0 1970-01-01 00:00:00.000000000+00:00 1970-01-01 00:00:00.000000001+00:00
1 1970-01-01 00:00:00.000000002+00:00 1970-01-01 00:00:00.000000003+00:00
2 1970-01-01 00:00:00.000000004+00:00 1970-01-01 00:00:00.000000005+00:00
3 1970-01-01 00:00:00.000000006+00:00 1970-01-01 00:00:00.000000007+00:00
4 1970-01-01 00:00:00.000000008+00:00 1970-01-01 00:00:00.000000009+00:00
```
#### output of `pd.show_versions()`

```
python: 3.5.1.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 58 Stepping 9, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.18.0rc1+66.gce3ac93
nose: 1.3.7
pip: 8.0.2
setuptools: 19.2
Cython: 0.23.4
numpy: 1.10.1
scipy: None
statsmodels: None
xarray: None
IPython: 4.0.2
sphinx: 1.3.1
patsy: None
dateutil: 2.4.2
pytz: 2015.7
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: 1.5.1
openpyxl: 2.3.3
xlrd: 0.9.4
xlwt: 1.0.0
xlsxwriter: None
lxml: 3.5.0
bs4: 4.4.1
html5lib: 0.999
httplib2: None
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.8
```

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #12513
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/internals/construction.py [hunk 1]
```diff
@@ -152,9 +152,17 @@ def init_ndarray(values, index, columns, dtype=None, copy=False):
         return arrays_to_mgr([values], columns, index, columns, dtype=dtype)
     elif is_extension_array_dtype(values) or is_extension_array_dtype(dtype):
         # GH#19157
+
+        if isinstance(values, np.ndarray) and values.ndim > 1:
+            # GH#12513 a EA dtype passed with a 2D array, split into
+            #  multiple EAs that view the values
+            values = [values[:, n] for n in range(values.shape[1])]
```
Comment (WillAyd) [inline pandas/core/internals/construction.py]: Do we only have to do this for the EA-case? Seems a little inefficient to have to do this here - not something that can handled a level deeper?

**Comment by WillAyd** [comment]
Can you add a whatsnew for this?

File: pandas/core/internals/construction.py [hunk 2]
```diff
@@ -152,9 +152,17 @@ def init_ndarray(values, index, columns, dtype=None, copy=False):
         return arrays_to_mgr([values], columns, index, columns, dtype=dtype)
     elif is_extension_array_dtype(values) or is_extension_array_dtype(dtype):
         # GH#19157
+
+        if isinstance(values, np.ndarray) and values.ndim > 1:
```
Comment (jreback) [inline pandas/core/internals/construction.py]: actually this is likely duplicating the category checks above

Comment (jreback) [reply pandas/core/internals/construction.py]: need to handle the copy arg

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/internals/construction.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/internals/construction.py]: > actually this is likely duplicating the category checks above

Yah we can probably do a grep for is_categorical_dtype and find a lot of things to clean up (adding this to my grep list...)

> need to handle the copy arg

how so?  copy is passed to prep_ndarray below, and so far all we've done is take views

File: pandas/core/internals/construction.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/internals/construction.py]: only for EA since they cant be 2D, correct

File: pandas/tests/frame/test_constructors.py [hunk 3]
```diff
@@ -2551,3 +2551,11 @@ def test_from_tzaware_mixed_object_array(self):
             "datetime64[ns, CET]",
         ]
         assert (res.dtypes == expected_dtypes).all()
+
+    def test_from_2d_ndarray_with_dtype(self):
```
Comment (jreback) [inline pandas/tests/frame/test_constructors.py]: can you make this test more generic and add to the EA tests instead (followup ok)

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. needs a rebase. comment on test location, can do here or in a followup

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks, followup (pls create an issue or add to list)

**Pull request merged**